packages/core/primitives/event-dispatch/src/dispatcher.ts

```diff
@@ -50,20 +39,26 @@ export class Dispatcher {
 
   /**
    * Options are:
-   *   1. `eventReplayer`: When the event contract dispatches replay events
+   *   - `eventReplayer`: When the event contract dispatches replay events
    *      to the Dispatcher, the Dispatcher collects them and in the next tick
-   *      dispatches them to the `eventReplayer`.
+   *      dispatches them to the `eventReplayer`. Defaults to dispatching to `dispatchDelegate`.
    * @param dispatchDelegate A function that should handle dispatching an `EventInfoWrapper` to handlers.
    */
   constructor(
     private readonly dispatchDelegate: (eventInfoWrapper: EventInfoWrapper) => void,
     {
-      actionResolver = undefined,
-      eventReplayer = undefined,
+      actionResolver,
+      eventReplayer,
     }: {actionResolver?: ActionResolver; eventReplayer?: Replayer} = {},
   ) {
     this.actionResolver = actionResolver;
-    this.eventReplayer = eventReplayer;
+    this.eventReplayer =
+      eventReplayer ??
+      ((eventInfoWrappers: EventInfoWrapper[]) => {
+        for (const eventInfoWrapper of eventInfoWrappers) {
+          this.dispatchDelegate(eventInfoWrapper);
+        }
+      });
   }
```

Comment:
nit: we can consider extracting a logic that creates a default event replayed to a separate function:
```suggestion
      eventReplayer ?? this.createDefaultEventReplayer();
  }
  
  /**
   * Creates a default event replayer, which is used if custom one is not provided.
   * This replayer dispatches all accumulated events using dispatch delegate logic
   * provided to the Dispatcher class at construction time.
   */
  private createDefaultEventReplayer() {
    return ((eventInfoWrappers: EventInfoWrapper[]) => {
      for (const eventInfoWrapper of eventInfoWrappers) {
        this.dispatchDelegate(eventInfoWrapper);
      }
    });
  }
```

Reply:
Extracted the logic in a slightly different way. I created a new factory called `createEventReplay`, and the default value of `eventReplayer` is now `createEventReplay(dispatchDelegate)`.